inlang/source-code/editor/src/renderer/_default.page.server.tsx

```diff
@@ -0,0 +1,106 @@
+import type { PageContextRenderer } from "./types.js"
+import { generateHydrationScript } from "solid-js/web"
+import { escapeInject, dangerouslySkipEscape } from "vike/server"
+import { setCurrentPageContext } from "./state.js"
+
+// import the css
+import "./app.css"
+import { renderTags } from "@solidjs/meta"
+import {
+	sourceLanguageTag,
+	availableLanguageTags,
+	languageTag,
+} from "@inlang/paraglide-js/inlang-marketplace"
+
+// See https://vite-plugin-ssr.com/data-fetching
+export const passToClient = ["pageProps", "routeParams", "languageTag"] as const
+
+export async function render(pageContext: PageContextRenderer): Promise<unknown> {
+	//! TODO most likely cross request state pollution
+	//! Need to look into this in the future
+	setCurrentPageContext(pageContext)
+	// generating the html from the server:
+	// 1. the server sends a hydration script for the client.
+	//    the client uses the hydration script to hydrate the page.
+	//    without hydration, no interactivity.
+	// 2. the page is pre-rendered via `renderedPage`.
+	//    pre-rendering the page makes the page immediately "visible"
+	//    to the user. Afterwards, the client hydrates the page and thereby
+	//    makes the page interactive.
+	// ! important: renderToString is used instead of
+	// ! renderToStringAsync some async resources should
+	// ! not be loaded on the server (the editor for example).
+	// ! see https://github.com/inlang/monorepo/issues/247
+
+	// from solidjs meta
+	// mutated during render so you can include in server-rendered template later
+	const tags: any[] = []
+
+	return escapeInject`<!DOCTYPE html>
+    <html lang="en" class="min-h-screen min-w-screen overflow-x-hidden">
+      <head>
+			<meta charset="UTF-8" />
+			<meta name="viewport" content="width=device-width, initial-scale=1.0" />
+		<!-- theme-color means the background color of the iOS status bar -->
+			<meta name="theme-color" content="#000000" />
+		<!-- START import inter font -->
+			<link rel="preconnect" href="https://rsms.me/">
+			<link rel="stylesheet" href="https://rsms.me/inter/inter.css">
+		<!-- END import inter font -->
+			${dangerouslySkipEscape(import.meta.env.PROD ? analytics : "")}
+			${dangerouslySkipEscape(favicons)}
+			${dangerouslySkipEscape(generateHydrationScript())}
```

Comment:
given that the editor should be CSR only, no hydration script is required. Same goes for the header tags. We can remove all of that

Reply:
I think we still need the file according to the [vike documentation](https://vike.dev/render-modes). Isn't the metadata still captured by Googles search engine?

I will remove the hydrationScript 👍🏼

Reply:
@samuelstroschein hydrationScript and tags are removed ✅ 
There is a Google analytics tag from 10 months ago. Can I remove this?